src/main/java/com/twilio/oai/resolver/node/NodeConventionResolver.java

```diff
@@ -0,0 +1,91 @@
+import org.openapitools.codegen.utils.StringUtils;
+
+import java.util.*;
+import java.util.function.Function;
+
+public class NodeConventionResolver {
+    final Map<String, Map<String, Object>> conventionMap;
+    static final String CONFIG_NODE_JSON_PATH = "config/node.json";
+
+    public NodeConventionResolver(){
+        conventionMap = getConventionalMap();
+    }
+
+    public Optional<CodegenParameter> resolveParameter(CodegenParameter codegenParameter) {
+        if (codegenParameter.dataFormat == null) {
+            return Optional.of(codegenParameter);
+        }
+        boolean hasProperty = conventionMap.get(Segments.SEGMENT_PROPERTIES.getSegment()).containsKey(StringUtils.underscore(codegenParameter.dataFormat));
+        if (hasProperty) {
+            codegenParameter.dataType = (String) conventionMap.get(Segments.SEGMENT_PROPERTIES.getSegment()).get(StringUtils.underscore(codegenParameter.dataFormat));
+        }
+        return Optional.of(codegenParameter);
+    }
+
+    public CodegenModel resolveModel(CodegenModel model){
+        for (CodegenProperty property : model.vars) {
+            if (property.dataFormat != null) {
+                boolean hasProperty = conventionMap.get(Segments.SEGMENT_PROPERTIES.getSegment()).containsKey(StringUtils.underscore(property.dataFormat));
+                if (hasProperty) {
+                    property.dataType = (String)conventionMap.get(Segments.SEGMENT_PROPERTIES.getSegment()).get(StringUtils.underscore(property.dataFormat));
+                }
+            }
+        }
+        return model;
+    }
+
+    public CodegenModel resolveComplexType(CodegenModel model, Map<String, String> modelFormatMap){
+        for (CodegenProperty prop: model.vars) {
+            if(modelFormatMap.containsKey(prop.complexType)) {
+                boolean hasProperty =  conventionMap.get(Segments.SEGMENT_PROPERTIES.getSegment()).
+                        containsKey(StringUtils.underscore(modelFormatMap.get(prop.complexType)));
+                if (hasProperty) {
+                    prop.dataType = (String)conventionMap.get(Segments.SEGMENT_PROPERTIES.getSegment()).
+                            get(StringUtils.underscore(modelFormatMap.get(prop.complexType)));
+
+                    if (prop.containerType != null && prop.containerType.equals("array")) {
+                        prop.dataType = "Array<" + prop.dataType + ">";
+                    }
+
+                    model.vendorExtensions.computeIfAbsent("x-imports", k -> new ArrayList<>());
+                    Map<String, String> obj = (Map<String, String>) conventionMap.get(Segments.SEGMENT_LIBRARY.getSegment()).
+                            get(StringUtils.underscore(modelFormatMap.get(prop.complexType)));
+                    ArrayList<Map<String, String>> imports = (ArrayList<Map<String, String>>) model.vendorExtensions.get("x-imports");
+                    Map<String, String> objectImport = new HashMap<>();
```

Comment:
`objectImport` initialization & name/path setting should be moved under line 65 (if import doesn't already exist)